### DISCUSSION TITLE
More streamlined "assign people to faces" workflow (with keyboard only)

### DISCUSSION THREAD

**Discussion by joachimtingvold** [photoprism/photoprism#1868] [Ideas]
There should probably be a way to assign faces to people without having to leave the keyboard (even between photos), in some kind of nice workflow.

Ideally such a workflow would (for each face) have a single view that shows/does all of the following;
* show the context of the face (i.e. as much of the whole picture as possible)
* show a zoomed-in/cropped face (basically what's currently being done)
* maybe some relevant metadata? (image name, date, country/location based on gps if any, etc)
* automatically activate the textfield where you can search for people/person (or create new), showing a list of matched people
* if #1865 is implemented, the list of people should be sorted based on metadata, with the most likely at the top
* be able to select from search results with arrows, and select with enter (basically what's currently being done)
* a way to ignore/remove face (i.e. unknown people in the background, etc), either via hotkey, or via typing a special name (i.e. "delete" or "ignore" or similar, depending on the number of relevant options)
* once a person has been assigned to the face, show similar faces (if/when #1595 is implemented)
* refresh page with a new face (all faces on same photo in order, before next photo), and activate textfield to repeat the process

With such a workflow, you could quickly assign person to faces, without going back-and-forth in the UI, and also without having to switch between the keyboard and mouse.

**Comment by graciousgrey** [comment]
Thank you for those ideas! 

You are referring to matching people to untagged faces and not to face clusters under People>New, correct?

I moved this to discussions as it is more a brainstorming than a clearly defined to do (for now). 

We already have an open issue for keyboard shortcuts: https://github.com/photoprism/photoprism/issues/1131 and a lot of tickets to improve the people functionalities: https://github.com/photoprism/photoprism/issues?q=is%3Aissue+is%3Aopen+people+in%3Atitle

We'll come back to this, once we focus on people again (currently working on multi-user support).

**Reply by joachimtingvold** [reply]
> I moved this to discussions as it is more a brainstorming than a clearly defined to do (for now).

No worries.

> You are referring to matching people to untagged faces and not to face clusters under People>New, correct?

Both, I guess? Every place where you have to "put name into text field to search for a person to assign a face or cluster". I'm only a few days into using PP, but so far there has also been a lot of false positives within an identified cluster. So just assigning the clusters to a person blindly is probably not a good idea either.

The same flow we're discussing here, could then potentially also be used to manually go through all faces in a cluster (for those that want to do that). I also notice that if you try to do this now, that is to click on a cluster, and get all associated images/faces, and then try to assign any one face to a different person (if it was a false positive), it will assign the person to all those photos (even the positive ones, since they are all linked/clustered). So there are definitely a lot of room for improvement in the "assign person to faces" workflow in general.

(FWIW I'm not complaining of the current state, these are just ideas that came to mind while using PP).

> We already have an open issue for keyboard shortcuts

Yes, and that will help, and would be part of a proper workflow, but it would not solve it completely.

**Comment by joachimtingvold** [comment]
A good example is the "People -> New" flow, that is `/people/new` (not the "Show all new faces" at the bottom). I assign a name to one of the clusters, but then I click on it to verify that the matches looks OK. That brings me to an `/all` query. I look through the photos, and when I'm done, I want to assign a person to the next cluster. Currently I now have to click "People -> New" again to get back (which is not really good UX).

**Comment by WesSec** [comment]
I would love to see a more streamlined workflow for tagging faces too.
